buildSrc/subprojects/kotlin-dsl/src/main/kotlin/build/GradleApiParameterNamesTransform.kt

```diff
@@ -64,11 +67,12 @@ fun Project.withCompileOnlyGradleApiModulesWithParameterNames(vararg gradleModul
     }
 
     dependencies {
-        registerTransform {
+        registerTransform(GradleApiParameterNamesTransform::class) {
```

Comment:
We definitely should have a reified version of this.

Reply:
I guess we should not for now, see this discussion on [Slack](https://gradle.slack.com/archives/C0WTFHWT1/p1550565252037400?thread_ts=1550564866.036600&cid=C0WTFHWT1):

> Paul Merlin   [2 days ago]
Using the plain Java API would give:
```registerTransform(Something::class.java) { ... }```
which isn’t nice.
> 
> But we also automatically generate `KClass` taking extensions for `Class` taking functions of the Gradle API. Meaning you can already do:
```registerTransform(Something::class) { ... }```
which is good enough.
> 
> Stefan Wolf   [2 days ago]
That is what I thought. So we only add the reified versions for frequently used APIs I guess?
>
> Paul Merlin   [2 days ago]
We also have plans to generate reified extension functions for the Gradle API, uniformly, but we won’t get to it soon.
>
> Paul Merlin   [2 days ago]
Exactly